File: go/problems/day5_test.go
package problems

import "testing"

var day5 Solvable[int64] = Day5{}

func TestDay5_TestSet_ShouldYield_Result(t *testing.T) {
	input := readProblemInput("day5_1")
	result := day5.Solve(input)

	var expected int64 = 35
	if result != expected {
		t.Errorf("expected %d but got %d", expected, result)
	}
}

func TestDay5_FullSet_ShouldYield_Result(t *testing.T) {
	input := readProblemInput("day5_2")
	result := day5.Solve(input)

	var expected int64 = 486613012
	if result != expected {
		t.Errorf("expected %d but got %d", expected, result)
	}
}

File: ts/problems/day5.test.ts
import test, { describe } from "node:test";
import assert from "node:assert";

import { solve } from "./day5.js";
import { readProblemInput } from "../inputReader.js";

describe("Day 5", () => {
  test("test set should yield expected result", () => {
    const input = readProblemInput("day5_1");
    const result = solve(input);
    assert.equal(result, 35);
  });

  test("full set should yield expected result", () => {
    const input = readProblemInput("day5_2");
    const result = solve(input);
    assert.equal(result, 486613012);
  });
});
